### DISCUSSION TITLE
Proxy HW

### DISCUSSION THREAD

**Discussion by DamijanD** [gekkekoe/esphome-ecodan-hp#70] [Announcements]
Need some help with buying HW to create this device with proxy support (https://github.com/gekkekoe/esphome-ecodan-hp/blob/main/proxy.md). So in the readme we have listed these parts:
- atom s3 lite https://www.digikey.nl/en/products/detail/m5stack-technology-co-ltd/C124/18070571
- grove cable (multiple options/lengths available) https://www.digikey.nl/en/products/detail/seeed-technology-co-ltd/110990036/5482563
- JST PAP-05V-S connector https://www.digikey.nl/en/products/detail/jst-sales-america-inc/PAP-05V-S/759977

Can please someone provide links (from the same shop if possible) for parts I need to have to to be able to have proxy option as well:
- pin header (4 pins? )
- "new CN105" socket (is this B05B-PASK-1 ? )

**Comment by gekkekoe** [comment]
@fonske can you help him out? 
he has made a plug and play pcb and should be able to help you

**Comment by fonske** [comment]
Hi @DamijanD 

I have made -ready to connect- PCBs with 1 mtr CN105 to CN105 cable and a Atom S3 lite.
Please contact me by email  alphonsuijtdehaag at gmail dot com for further details/address etc if you are interested.
Or send me a PM. Or buy from Tindie: https://www.tindie.com/products/fontecservice/mitsubishi-ecodan-hp-air-to-water-proxy-pcb/

For UK based users, please contact @F1p 

![proxy2](https://github.com/user-attachments/assets/c4b9f43b-8741-4eb7-a17c-a45e0604742d)

**Reply by JRascagneres** [reply]
I have emailed you to ask about the PCB too :)

**Reply by clokey** [reply]
Hi @F1p i am in the uk, how can I get in touch with you about getting a prebuilt proxy? Would love to get this in place before the cold weather arrives

**Reply by F1p** [reply]
> Hi @F1p i am in the uk, how can I get in touch with you about getting a prebuilt proxy? Would love to get this in place before the cold weather arrives

Hey,
Tried to contact you - hope it has worked!

Edit - it bounced, here i am:
philip.thomson at f1p.uk

**Comment by gekkekoe** [comment]
~~for ppl with proxy setup, can you please test https://github.com/gekkekoe/esphome-ecodan-hp/tree/proxy_with_fallback
and see if it works properly ? (you can disconnect the slave, and it should take over, connect the slave and the slave is in command again).~~

It's basically plug and play proxy.

**Reply by DamijanD** [reply]
So this means that I can connect or disconnect the slave during powered HP and everything should work? (what about master is it save to connect or disconnect it while HP is powered on?)
I'll test this when I'll be finished with WP3 modifications.

**Comment by gekkekoe** [comment]
Correct, the master maintains the connection and will take over (query) when melcloud did not send any response within 2 mins interval

When melcloud is reconnected, the master backs off. So it should make it easier to work with proxy (when melcloud is down for example)

**Comment by DamijanD** [comment]
@gekkekoe I have tested this proxy with fallback and it looks like it is working. What I did:
- installed new version
- changed the room temp via melcloud 
- disconnected wifi module for 2+ minutes (esphome was still working and comunicating)
- reconnected wifi module, waited for 2+ minutes and again changed the room temp via melcloud
- in both "room change temp commands" everything was ok.

**Reply by gekkekoe** [reply]
thanks for confirming!
